<launch>

    <!--stand ros unit m | m/s | rad-->
    <arg name="map_size_x" default="40.0"/>
    <arg name="map_size_y" default="40.0"/>
    <arg name="map_size_z" default=" 0.2"/>

    <!--map origin default = 1/2 map size-->
    <arg name="map_orign_x" default="20.0"/>
    <arg name="map_orign_y" default="20.0"/>
    <arg name="map_orign_z" default=" 0.1"/>


    <arg name="start_x" default=" 0.0"/>
    <arg name="start_y" default=" 0.0"/>
    <arg name="start_z" default=" 0.0"/>

    <arg name="start_vx" default=" 0.2"/>
    <arg name="start_vy" default=" 0.2"/>
    <arg name="start_vz" default=" 0.0"/>

    <arg name="map_name" default="map11.png"/>

    <arg name="erode_kernel_size" default="30.0"/>
    <arg name="erode_switch" default="true"/>

    <node pkg ="grid_path_searcher" name ="image_map" type ="image_map" output = "screen">

        <remap from="~odom" to="/ground_truth/state"/>

        <param name="init_state_x" value="$(arg start_x)"/>
        <param name="init_state_y" value="$(arg start_y)"/>

        <param name="map/x_size" value="$(arg map_size_x)" />
        <param name="map/y_size" value="$(arg map_size_y)" />
        <param name="map/z_size" value="$(arg map_size_z)" />

        <param name="map/orign_x_size" value="$(arg map_orign_x)"/>
        <param name="map/orign_y_size" value="$(arg map_orign_y)"/>
        <param name="map/orign_z_size" value="$(arg map_orign_z)"/>

        <param name="map/local_w" value="4"/>
        <param name="map/resolution" value="0.05"/>
        <param name="map/erode_kernel_size" value="$(arg erode_kernel_size)" />
        <param name="map/erode_switch" value="$(arg erode_switch)" />

        <param name="sensing/rate" value="10"/>

        <param name="image_address" value="$(find grid_path_searcher)/map/$(arg map_name) "/>
    </node>

    <node pkg="waypoint_generator" name="waypoint_generator" type="waypoint_generator" output="screen">
        <remap from="~goal" to="/goal"/>
        <param name="waypoint_type" value="manual-lonely-waypoint"/>
    </node>
    <!-- swift_rviz.rviz -->
    <node name="img_rviz" pkg="rviz" type="rviz" args="-d $(find grid_path_searcher)/launch/rviz_config/planner_test.rviz"/>

    <node pkg="tf" type="static_transform_publisher" name="tf_world_to_map" 
        args="0 0 0 0 0 0 1 odom world 100 " />
    <node pkg="tf" type="static_transform_publisher" name="tf_planner_to_world" 
        args="0 0 0 0 0 0 1 world planner 100 " />
    <!--static_transform_publisher x y z yaw pitch roll frame_id child_frame_id-->
    <!--static_transform_publisher x y z qx qy qz qw frame_id child_frame_id-->
</launch>
